### PR TITLE
Farming section and move Nova down Domain section

### PR DESCRIPTION
**Pull request by marc-aurele-besner**
## Farming section and move Nova down Domain section

- Add a section to capture farming, the download of Space Acres and more
- Add link to docks for CLI
- Automatically fetch latest github release and serve the right os link to user
- Show a Domain section with both Nova and Auto-Id

### Screenshots

Farming
Farming-2
Farming-3

Domain

Auto-Id

___

### **PR Type**
Enhancement, Documentation

___

### **Description**
- Added new layout and page components for the domain and farming routes.
- Introduced `DownloadPage` component for farming with OS-specific download links and instructions.
- Updated `OutOfSyncBanner` to exclude domain routes.
- Added `FarmingHeader` and updated `DomainHeader` with new menu structure and theme toggle.
- Renamed `DomainHeader` to `SectionHeader` in `MainLayout`.
- Updated routes to include new farming and domain routes with nested domain routes.
- Updated Next.js configuration to allow importing images from the docs domain.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><details><summary>11 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>layout.tsx</strong><dd><code>Add layout component for domain route</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

explorer/src/app/[chain]/domain/layout.tsx

<li>Added a new layout component for the domain route.<br> <li> Utilizes <code>MainLayout</code> with <code>DomainHeader</code> as a sub-header.<br>

</details>

  </td>
  <td><a href="https://github.com/subspace/astral/pull/754/files#diff-1010327144c96f2c0643eecc4cc9cdcb4687c1517c51d18e6760966f700581c5">+7/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>page.tsx</strong><dd><code>Add page component for domain route</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

explorer/src/app/[chain]/domain/page.tsx

<li>Added a new page component for the domain route.<br> <li> Generates metadata for the domain page.<br>

</details>

  </td>
  <td><a href="https://github.com/subspace/astral/pull/754/files#diff-be79d2a19b6254ec3fde7487c5c9ab8172fb180bd905d0f827eba67414600c04">+28/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>layout.tsx</strong><dd><code>Add layout component for farming route</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

explorer/src/app/[chain]/farming/layout.tsx

<li>Added a new layout component for the farming route.<br> <li> Utilizes <code>MainLayout</code> with <code>FarmingHeader</code> as a sub-header.<br>

</details>

  </td>
  <td><a href="https://github.com/subspace/astral/pull/754/files#diff-58528c751f7650f1631fa8239678b9d33a73479036701546c1e58a9c0547acb0">+7/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>page.tsx</strong><dd><code>Add page component for farming route</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

explorer/src/app/[chain]/farming/page.tsx

<li>Added a new page component for the farming route.<br> <li> Generates metadata for the farming page.<br>

</details>

  </td>
  <td><a href="https://github.com/subspace/astral/pull/754/files#diff-5129abc95641f6261a8e358efd8be7a3426e58dedb209f69696f30ee40834dcb">+29/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>index.tsx</strong><dd><code>Add DownloadPage component for farming</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

explorer/src/components/Farming/index.tsx

<li>Added a new <code>DownloadPage</code> component for farming.<br> <li> Fetches latest release assets from GitHub.<br> <li> Provides OS-specific download links and instructions.<br>

</details>

  </td>
  <td><a href="https://github.com/subspace/astral/pull/754/files#diff-56a294ba09aa34bd7c8432524904ac828bb612a22268b3ab7388d52430f22468">+251/-0</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>OutOfSyncBanner.tsx</strong><dd><code>Update OutOfSyncBanner to exclude domain routes</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

explorer/src/components/common/OutOfSyncBanner.tsx

<li>Updated <code>useOutOfSyncBanner</code> to exclude domain routes from showing the <br>out-of-sync banner.<br>

</details>

  </td>
  <td><a href="https://github.com/subspace/astral/pull/754/files#diff-4babc9c3a2f6222cf07962354c5e6fcc136efad5de58d0096229ab9b1c74e13d">+2/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>DomainHeader.tsx</strong><dd><code>Update DomainHeader with new menu structure and theme toggle</code></dd></summary>
<hr>

explorer/src/components/layout/DomainHeader.tsx

<li>Updated <code>DomainHeader</code> component to use new menu structure.<br> <li> Added theme toggle functionality.<br>

</details>

  </td>
  <td><a href="https://github.com/subspace/astral/pull/754/files#diff-56fc61d952332daad447cd085c4269054eda3a1fb8bb2503c750fbd2b99449af">+101/-84</a></td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>FarmingHeader.tsx</strong><dd><code>Add FarmingHeader component</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

explorer/src/components/layout/FarmingHeader.tsx

<li>Added a new <code>FarmingHeader</code> component for the farming route.<br> <li> Includes menu items and theme toggle functionality.<br>

</details>

  </td>
  <td><a href="https://github.com/subspace/astral/pull/754/files#diff-636dd5326906c981373c809fa16660e4cfc9af4196d782fc503999a07ccecb98">+121/-0</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>Layout.tsx</strong><dd><code>Rename DomainHeader to SectionHeader in MainLayout</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

explorer/src/components/layout/Layout.tsx

- Renamed `DomainHeader` to `SectionHeader` in `MainLayout`.

</details>

  </td>
  <td><a href="https://github.com/subspace/astral/pull/754/files#diff-d3a55ba608ad601d6b87f935e74bdef795050af0998deb7aac09a669d568c41e">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>SectionHeader.tsx</strong><dd><code>Add SectionHeader component</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

explorer/src/components/layout/SectionHeader.tsx

- Added a new `SectionHeader` component to replace `DomainHeader`.

</details>

  </td>
  <td><a href="https://github.com/subspace/astral/pull/754/files#diff-823ceb7cf6638c79eed2887ca8fc8d4d16e955deeaad8066d9f244c44cd0f635">+100/-0</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>routes.ts</strong><dd><code>Update routes to include farming and domain</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

explorer/src/constants/routes.ts

<li>Updated routes to include new farming and domain routes.<br> <li> Added nested routes for domain.<br>

</details>

  </td>
  <td><a href="https://github.com/subspace/astral/pull/754/files#diff-1db40c1c0135e9c55c149a79c123aafb3fc2408f35965369fe6ff07598b37d91">+23/-6</a>&nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr><tr><td><strong>Configuration changes</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>next.config.js</strong><dd><code>Update Next.js config for remote image imports</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

explorer/next.config.js

<li>Updated Next.js configuration to allow importing images from the docs <br>domain.<br>

</details>

  </td>
  <td><a href="https://github.com/subspace/astral/pull/754/files#diff-323d76f1a4cced7f820eb2c7bf11a396b621e96ead90ba71c337d7346572404f">+11/-0</a>&nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: explorer/src/components/Farming/index.tsx [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+    switch (userOS) {
+      case 'Windows':
+        return (
+          <div className='text-center'>
+            <h2 className='text-2xl font-bold'>Download Space Acres for Windows</h2>
+            <p>Minimum Requirements:</p>
+            <ul className='list-inside list-disc'>
+              <li>OS: Windows 10 or higher</li>
+              <li>Processor: Intel Core i3 or equivalent</li>
+              <li>RAM: 4 GB</li>
```
Comment (EmilFattakhov) [inline explorer/src/components/Farming/index.tsx:92]: Let's use the same min requirements as per [farming docs](https://docs.subspace.network/docs/farming-&-staking/farming/prerequisites), please:

```suggestion
              <li>RAM: 8 GB</li>
```

File: explorer/src/components/Farming/index.tsx [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+              <li>Disk Space: 500 MB</li>
```
Comment (EmilFattakhov) [inline explorer/src/components/Farming/index.tsx:93]: The app takes 500 MB, but when you actually start it, with the node it grows to tens of GB. I'd suggest to state 100GB here. 
```suggestion
              <li>Disk Space: 100 GB</li>
```

File: explorer/src/components/Farming/index.tsx [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+
+    switch (userOS) {
+      case 'Windows':
+        return (
+          <div className='text-center'>
+            <h2 className='text-2xl font-bold'>Download Space Acres for Windows</h2>
+            <p>Minimum Requirements:</p>
+            <ul className='list-inside list-disc'>
+              <li>OS: Windows 10 or higher</li>
+              <li>Processor: Intel Core i3 or equivalent</li>
```
Comment (EmilFattakhov) [inline explorer/src/components/Farming/index.tsx:91]: For processors, macOS users can get confused as i3 macs has been unavailable for quite a while. Let's just put 4 Core CPU for the processor for all systems (this one and all the ones below)
```suggestion
              <li>CPU: 4 Core </li>
```

File: explorer/src/components/Farming/index.tsx [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+            />
+          </div>
+        </div>
+
+        <div className='m-4 p-4'>
+          <h3 className='mb-8 mt-6 text-center text-4xl font-bold text-gray-900 dark:text-white'>
+            By contributing storage and compute to the network, you play a crucial role in securing
+            it, while also earning rewards.
+          </h3>
+          <h3 className='mb-8 mt-6 text-center text-4xl font-bold text-gray-900 dark:text-white'>
```
Comment (EmilFattakhov) [inline explorer/src/components/Farming/index.tsx:201]:
```suggestion
         
```

File: explorer/src/components/Farming/index.tsx [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+            Your work is recognized and compensated based on your active participation, not through
```
Comment (EmilFattakhov) [inline explorer/src/components/Farming/index.tsx:202]:
```suggestion

```

File: explorer/src/components/Farming/index.tsx [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+            passive handouts. This approach ensures a self-sustaining and equitable ecosystem.
```
Comment (EmilFattakhov) [inline explorer/src/components/Farming/index.tsx:203]:
```suggestion

```

File: explorer/src/components/Farming/index.tsx [hunk 7]
```diff
@@ -0,0 +1,240 @@
...
+          </h3>
```
Comment (EmilFattakhov) [inline explorer/src/components/Farming/index.tsx:204]:
```suggestion
          
```

File: explorer/src/components/Farming/index.tsx (hunk 4 above)
Comment (EmilFattakhov) [reply explorer/src/components/Farming/index.tsx:201]: I had a second thought that we shouldn't put this wording in the first version :)

**Review by EmilFattakhov — COMMENTED** [review]
Great idea Marc, left a few comments!

File: explorer/src/components/Domain/index.tsx [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+    [selectedChain.urls.page],
+  )
+
+  return (
+    <div className='flex w-full flex-col items-center space-y-4'>
+      <div className='w-full max-w-4xl'>
+        <div className='mb-4 w-full rounded-[20px] border border-slate-100 bg-white px-3 py-4 shadow dark:border-none dark:bg-gradient-to-r dark:from-gradientTwilight dark:via-gradientDusk dark:to-gradientSunset sm:p-6'>
+          <div className='mb-10 flex flex-col items-center justify-center'>
+            <h1 className='mb-8 mt-6 text-center text-4xl font-bold text-gray-900 dark:text-white'>
+              Domain
```
Comment (jfrank-summit) [inline explorer/src/components/Domain/index.tsx:39]: I would make this plural.
```suggestion
              Domains
```

File: explorer/src/constants/routes.ts [hunk 9]
```diff
@@ -1,8 +1,11 @@
 export enum Routes {
-  nova = 'nova',
   consensus = 'consensus',
-  leaderboard = 'leaderboard',
+  farming = 'farming',
   staking = 'staking',
+  leaderboard = 'leaderboard',
+  domain = 'domain',
```
Comment (jfrank-summit) [inline explorer/src/constants/routes.ts:6]: I would make this plural
```suggestion
  domains = 'domains',
```

File: explorer/src/constants/routes.ts [hunk 10]
```diff
@@ -13,16 +16,30 @@ export const ROUTES = [
...
   },
   {
     name: Routes.leaderboard,
     title: 'Leaderboard',
   },
   {
-    name: Routes.staking,
-    title: 'Staking',
+    name: Routes.domain,
+    title: 'Domain',
```
Comment (jfrank-summit) [inline explorer/src/constants/routes.ts:32]:
```suggestion
    title: 'Domains',
```

**Review by jfrank-summit — COMMENTED** [review]
Looks great. I had a few comments about pluralizing "domain".

**Review by jfrank-summit — APPROVED** [review]

**Pull request merged**